site/blog/2022/nodejs-deploy-codepipeline.mdx

```diff
@@ -0,0 +1,221 @@
+  FROM subnet
+  WHERE availability_zone = 'us-east-1a'
+  LIMIT 1;
+
+INSERT INTO instance_security_groups (instance_id, security_group_id) SELECT
+  (SELECT id FROM instance WHERE tags ->> 'name' = 'codedeploy-app'),
+  (SELECT id FROM security_group WHERE group_name='codedeploy');
+
+SELECT * FROM iasql_commit();
+```
+
+This fragment will install the required modules, and all the infrastructure needed to deploy the pipeline, including the needed Roles, Security Rules, Storage and Compute on EC2. After this process finishes, the pipeline is ready to go. It can be launched in another query (please remember to replace the _token_ with your current one):
+
+```sql TheButton[Trigger CodePipeline]="Trigger CodePipeline"
+SELECT * FROM iasql_begin();
+
+INSERT INTO codedeploy_application (name, compute_platform)
+VALUES ('codedeploy-app', 'Server');
+
+INSERT INTO codedeploy_deployment_group (application_id, name, role_name, ec2_tag_filters)
+VALUES ((SELECT id FROM codedeploy_application WHERE name = 'codedeploy-app'), 'dg-name', 'codedeploy_role', '[
+{
+"Type": "KEY_AND_VALUE",
+"Key": "name",
+"Value": "codedeploy-app"
+}
+]');
+
+INSERT INTO pipeline_declaration (name, service_role_name, stages, artifact_store)
+VALUES ('codedeploy_pipeline', 'codepipeline_role', '[
+{
+"name": "Source",
+"actions": [
+  {
+    "name": "SourceAction",
+    "actionTypeId": {
+      "category": "Source",
+      "owner": "ThirdParty",
+      "version": "1",
+      "provider": "GitHub"
+    },
+    "configuration": {
+      "Owner": "iasql",
+      "Repo": "iasql-code-example-node",
+      "Branch": "main",
+      "OAuthToken": "<your_personal_token>"
+    },
+    "outputArtifacts": [
+      {
+        "name": "Source"
+      }
+    ]
+  }
+]
+},
+{
+"name": "Deploy",
+"actions": [
+  {
+    "name": "DeployApp",
+    "actionTypeId": {
+      "category": "Deploy",
+      "owner": "AWS",
+      "version": "1",
+      "provider": "CodeDeploy"
+    },
+    "configuration": {
+      "ApplicationName": "codedeploy-app",
+      "DeploymentGroupName": "dg-name"
+    },
+    "inputArtifacts": [
+      {
+        "name": "Source"
+      }
+    ]
+  }
+]
+}
+]', '{ "type": "S3", "location": "codedeploy-app-iasql" }');
+
+SELECT * FROM iasql_commit();
+```
+
+This pipeline is describing all the needed steps to deploy the application:
+- **Source** defines where to pick the source code from, and where to store it. It will pick from the defined GitHub repo, and it will store on the S3 bucket defined in the artifact configuration
+- ** Deploy ** defines the action to take. It will trigger a deployment based on the created application and the specific deployment group.
+
+The status of the pipeline can be checked via AWS console:
+
+<img src="/img/codepipeline_status.png" style={{ maxWidth: 600 }} />
+
+
+After an small amount of time the pipeline will run and the application will be ready to go. It will be exposed on the EC2 public ip, that can be extracted with:
```

Comment:
```suggestion
After a small amount of time the pipeline will run and the application will be ready to go. It will be exposed on the EC2 public ip, that can be extracted with:
```